### ISSUE TITLE
Is the testing char appeared during training

### ISSUE THREAD

**Issue by Godricly** [kaonashi-tyc/zi2zi#6]
Hi, I'm playing with your code and it looks great. Meanwhile I'm following cycleGAN to do style transform  too.  I'd like to ask you several some questions, your reply is appreciated.
1. Is the char for testing appear in training? Are they memorized by the model? 
2. How do you train 27 different fonts at same time? It seems that the image generating only supports 2 types of font.
3.  How are  the Gaussian noises mapped to the font type? Is the category loss predicting the font type?
4.  Your work is trained using pairs of images. Can I view it as a semi-supervised approach?
5.  If I add noises(Gaussian) to the background of images, can your algorithm tolerate these noises?

Long may the sun shine! 🌞

**Comment by kaonashi-tyc** [comment]
1. The examples appear in comparison with ground truth are held out from the model during training.
2. In font2img.py, there is a label option, that label will be used as the font label during the training. Noted that after you run font2img.py, you need to run package.py as well, package.py will read the label from the name of images generated by font2img.py.
3. According to the label you assigned during the image generation.
4. At this point it is actually supervised. However it could be changed into semi-supervised version, with some trick of the loss function.
5. That I don't know, you need to try it out.

**Comment by Godricly** [comment]
Thank you so much.
